### PR TITLE
ClientID to Mint argument

### PR DESCRIPTION
**Pull request by Dmdv**
### Linked update on the smart contract
https://github.com/0chain/token_bridge_SC/blob/master/contracts/Bridge.sol#L74

image

Previous implementation of nonce tracking
```solidity
    mapping(uint256) private userNonceMinted;
```

Current implementation of nonce tracking
```solidity
    // Mapping client -> nonce when minted where the nonce is the burn nonce in ZCN
    mapping(bytes => uint256) private userNonceMinted;
```

### Changes

- Updated mint function in Bridge solidity contract
- Updated solidity Go bindings
- Removed ID() - not working
- Added ClientID()
- Added clientID parameter to nonce
- Corrected minor fixes
- Added linting
- Added formatting

### Fixes

- Less gas consumption
- Less state size
- Fixed burn nonce check during mint operation

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
https://github.com/0chain/0chain/pull/1402

- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: zcnbridge/example/main.go [hunk 1]
```diff
@@ -1,6 +1,3 @@
-//go:build ignore
```
Comment (dabasov) [inline zcnbridge/example/main.go]: do we need to build this?

**Review by dabasov — CHANGES_REQUESTED** [review]

File: zcnbridge/example/main.go (hunk 1 above)
Comment (Dmdv) [reply zcnbridge/example/main.go]: Added build constraints

Comment (Dmdv) [reply zcnbridge/example/main.go]: Done

**Review by dabasov — APPROVED** [review]
lgtm

**Review by dabasov — COMMENTED** [review]
lgtm

**Pull request merged**